accio-base/src/main/java/io/accio/base/client/duckdb/DuckdbClient.java

```diff
@@ -225,4 +250,51 @@ public void close()
             throw new RuntimeException(e);
         }
     }
+
+    public void closeAndInitPool()
```

Comment:
```suggestion
    public synchronized void closeAndInitPool()
```
Let's `synchronized` this method to be thread safe.